From 859918c22b349ab1a4c2274d0ccaa05f42f85e08 Mon Sep 17 00:00:00 2001
From: Karthikeyan Periyasamy <periyasa@codeaurora.org>
Date: Tue, 16 Jul 2019 19:32:50 +0530
Subject: [PATCH] ath11k: rename the data_lock for better code

Having same lock name in two different structure lead to
confuse the code readability. Avoid by renaming one of
the lock name as base_lock.

Signed-off-by: Karthikeyan Periyasamy <periyasa@codeaurora.org>
Signed-off-by: Kalle Valo <kvalo@codeaurora.org>
---
 drivers/net/wireless/ath/ath11k/core.c        |  6 ++---
 drivers/net/wireless/ath/ath11k/core.h        |  2 +-
 drivers/net/wireless/ath/ath11k/debug.c       |  4 ++--
 drivers/net/wireless/ath/ath11k/debugfs_sta.c | 16 ++++++-------
 drivers/net/wireless/ath/ath11k/dp.c          |  6 ++---
 drivers/net/wireless/ath/ath11k/dp_rx.c       | 34 +++++++++++++--------------
 drivers/net/wireless/ath/ath11k/mac.c         | 30 +++++++++++------------
 drivers/net/wireless/ath/ath11k/peer.c        | 28 +++++++++++-----------
 drivers/net/wireless/ath/ath11k/reg.c         |  6 ++---
 drivers/net/wireless/ath/ath11k/wmi.c         |  8 +++----
 10 files changed, 70 insertions(+), 70 deletions(-)

--- a/drivers/net/wireless/ath/ath11k/core.c
+++ b/drivers/net/wireless/ath/ath11k/core.c
@@ -713,9 +713,9 @@ static void ath11k_core_restart(struct w
 	struct ath11k_pdev *pdev;
 	int i, ret = 0;
 
-	spin_lock_bh(&sc->data_lock);
+	spin_lock_bh(&sc->base_lock);
 	sc->stats.fw_crash_counter++;
-	spin_unlock_bh(&sc->data_lock);
+	spin_unlock_bh(&sc->base_lock);
 
 	for (i = 0; i < sc->num_radios; i++) {
 		pdev = &sc->pdevs[i];
@@ -879,7 +879,7 @@ struct ath11k_base *ath11k_core_alloc(st
 #endif
 
 	mutex_init(&sc->core_lock);
-	spin_lock_init(&sc->data_lock);
+	spin_lock_init(&sc->base_lock);
 
 	INIT_LIST_HEAD(&sc->peers);
 	init_waitqueue_head(&sc->peer_mapping_wq);
--- a/drivers/net/wireless/ath/ath11k/core.h
+++ b/drivers/net/wireless/ath/ath11k/core.h
@@ -710,7 +710,7 @@ struct ath11k_base {
 	/* To synchronize core_start/core_stop */
 	struct mutex core_lock;
 	/* Protects data like peers */
-	spinlock_t data_lock;
+	spinlock_t base_lock;
 	struct ath11k_pdev pdevs[MAX_RADIOS];
 	struct ath11k_pdev __rcu *pdevs_active[MAX_RADIOS];
 	struct ath11k_hal_reg_capabilities_ext hal_reg_cap[MAX_RADIOS];
--- a/drivers/net/wireless/ath/ath11k/debugfs_sta.c
+++ b/drivers/net/wireless/ath/ath11k/debugfs_sta.c
@@ -160,11 +160,11 @@ void ath11k_update_per_peer_stats_from_t
 	int ret;
 
 	rcu_read_lock();
-	spin_lock_bh(&ab->data_lock);
+	spin_lock_bh(&ab->base_lock);
 	peer = ath11k_peer_find_by_id(ab, ts->peer_id);
 	if (!peer || !peer->sta) {
 		ath11k_warn(ab, "failed to find the peer\n");
-		spin_unlock_bh(&ab->data_lock);
+		spin_unlock_bh(&ab->base_lock);
 		rcu_read_unlock();
 		return;
 	}
@@ -181,7 +181,7 @@ void ath11k_update_per_peer_stats_from_t
 							    &rate_idx,
 							    &rate);
 		if (ret < 0) {
-			spin_unlock_bh(&ab->data_lock);
+			spin_unlock_bh(&ab->base_lock);
 			rcu_read_unlock();
 			return;
 		}
@@ -189,7 +189,7 @@ void ath11k_update_per_peer_stats_from_t
 	} else if (ts->pkt_type == HAL_TX_RATE_STATS_PKT_TYPE_11N) {
 		if (ts->mcs > 7) {
 			ath11k_warn(ab, "Invalid HT mcs index %d\n", ts->mcs);
-			spin_unlock_bh(&ab->data_lock);
+			spin_unlock_bh(&ab->base_lock);
 			rcu_read_unlock();
 			return;
 		}
@@ -201,7 +201,7 @@ void ath11k_update_per_peer_stats_from_t
 	} else if (ts->pkt_type == HAL_TX_RATE_STATS_PKT_TYPE_11AC) {
 		if (ts->mcs > 9) {
 			ath11k_warn(ab, "Invalid VHT mcs index %d\n", ts->mcs);
-			spin_unlock_bh(&ab->data_lock);
+			spin_unlock_bh(&ab->base_lock);
 			rcu_read_unlock();
 			return;
 		}
@@ -218,7 +218,7 @@ void ath11k_update_per_peer_stats_from_t
 	arsta->txrate.bw = ath11k_mac_bw_to_mac80211_bw(ts->bw);
 
 	ath11k_accumulate_per_peer_tx_stats(arsta, peer_stats, rate_idx);
-	spin_unlock_bh(&ab->data_lock);
+	spin_unlock_bh(&ab->base_lock);
 	rcu_read_unlock();
 }
 
@@ -349,7 +349,7 @@ static ssize_t ath11k_dbg_sta_dump_rx_st
 		return -ENOMEM;
 
 	mutex_lock(&ar->conf_mutex);
-	spin_lock_bh(&ar->ab->data_lock);
+	spin_lock_bh(&ar->ab->base_lock);
 
 	len += scnprintf(buf + len, size - len, "RX peer stats:\n");
 	len += scnprintf(buf + len, size - len, "Num of MSDUs: %llu\n",
@@ -471,7 +471,7 @@ static ssize_t ath11k_dbg_sta_dump_rx_st
 
 	len += scnprintf(buf + len, size - len, "\n");
 
-	spin_unlock_bh(&ar->ab->data_lock);
+	spin_unlock_bh(&ar->ab->base_lock);
 
 	if (len > size)
 		len = size;
@@ -632,9 +632,9 @@ static ssize_t ath11k_dbg_sta_reset_rx_s
 	if (!reset || reset > 1)
 		return -EINVAL;
 
-	spin_lock_bh(&ar->ab->data_lock);
+	spin_lock_bh(&ar->ab->base_lock);
 	memset(arsta->rx_stats, 0, sizeof(*arsta->rx_stats));
-	spin_unlock_bh(&ar->ab->data_lock);
+	spin_unlock_bh(&ar->ab->base_lock);
 
 	ret = count;
 	return ret;
--- a/drivers/net/wireless/ath/ath11k/dp.c
+++ b/drivers/net/wireless/ath/ath11k/dp.c
@@ -24,17 +24,17 @@ void ath11k_dp_peer_cleanup(struct ath11
 
 	/* TODO: Any other peer specific DP cleanup */
 
-	spin_lock_bh(&ab->data_lock);
+	spin_lock_bh(&ab->base_lock);
 	peer = ath11k_peer_find(ab, vdev_id, addr);
 	if (!peer) {
 		ath11k_warn(ab, "failed to lookup peer %pM on vdev %d\n",
 			    addr, vdev_id);
-		spin_unlock_bh(&ab->data_lock);
+		spin_unlock_bh(&ab->base_lock);
 		return;
 	}
 
 	ath11k_peer_rx_tid_cleanup(ar, peer);
-	spin_unlock_bh(&ab->data_lock);
+	spin_unlock_bh(&ab->base_lock);
 }
 
 int ath11k_dp_peer_setup(struct ath11k *ar, int vdev_id, const u8 *addr)
--- a/drivers/net/wireless/ath/ath11k/dp_rx.c
+++ b/drivers/net/wireless/ath/ath11k/dp_rx.c
@@ -705,7 +705,7 @@ static void ath11k_dp_rx_tid_mem_free(st
 	struct ath11k_peer *peer;
 	struct dp_rx_tid *rx_tid;
 
-	spin_lock_bh(&ab->data_lock);
+	spin_lock_bh(&ab->base_lock);
 
 	peer = ath11k_peer_find(ab, vdev_id, peer_mac);
 	if (!peer) {
@@ -724,7 +724,7 @@ static void ath11k_dp_rx_tid_mem_free(st
 	rx_tid->active = false;
 
 unlock_exit:
-	spin_unlock_bh(&ab->data_lock);
+	spin_unlock_bh(&ab->base_lock);
 }
 
 int ath11k_peer_rx_tid_setup(struct ath11k *ar, const u8 *peer_mac, int vdev_id,
@@ -739,12 +739,12 @@ int ath11k_peer_rx_tid_setup(struct ath1
 	dma_addr_t paddr;
 	int ret;
 
-	spin_lock_bh(&ab->data_lock);
+	spin_lock_bh(&ab->base_lock);
 
 	peer = ath11k_peer_find(ab, vdev_id, peer_mac);
 	if (!peer) {
 		ath11k_warn(ab, "failed to find the peer to set up rx tid\n");
-		spin_unlock_bh(&ab->data_lock);
+		spin_unlock_bh(&ab->base_lock);
 		return -ENOENT;
 	}
 
@@ -754,7 +754,7 @@ int ath11k_peer_rx_tid_setup(struct ath1
 		paddr = rx_tid->paddr;
 		ret = ath11k_peer_rx_tid_reo_update(ar, peer, rx_tid,
 						    ba_win_sz, ssn);
-		spin_unlock_bh(&ab->data_lock);
+		spin_unlock_bh(&ab->base_lock);
 		if (ret) {
 			ath11k_warn(ab, "failed to update reo for rx tid %d\n", tid);
 			return ret;
@@ -783,7 +783,7 @@ int ath11k_peer_rx_tid_setup(struct ath1
 
 	vaddr = kzalloc(hw_desc_sz + HAL_LINK_DESC_ALIGN - 1, GFP_ATOMIC);
 	if (!vaddr) {
-		spin_unlock_bh(&ab->data_lock);
+		spin_unlock_bh(&ab->base_lock);
 		return -ENOMEM;
 	}
 
@@ -796,7 +796,7 @@ int ath11k_peer_rx_tid_setup(struct ath1
 
 	ret = dma_mapping_error(ab->dev, paddr);
 	if (ret) {
-		spin_unlock_bh(&ab->data_lock);
+		spin_unlock_bh(&ab->base_lock);
 		goto err_mem_free;
 	}
 
@@ -805,7 +805,7 @@ int ath11k_peer_rx_tid_setup(struct ath1
 	rx_tid->size = hw_desc_sz;
 	rx_tid->active = true;
 
-	spin_unlock_bh(&ab->data_lock);
+	spin_unlock_bh(&ab->base_lock);
 
 	ret = ath11k_wmi_peer_rx_reorder_queue_setup(ar, vdev_id, peer_mac,
 						     paddr, tid, 1, ba_win_sz);
@@ -851,12 +851,12 @@ int ath11k_dp_rx_ampdu_stop(struct ath11
 	bool active;
 	int ret;
 
-	spin_lock_bh(&ab->data_lock);
+	spin_lock_bh(&ab->base_lock);
 
 	peer = ath11k_peer_find(ab, vdev_id, params->sta->addr);
 	if (!peer) {
 		ath11k_warn(ab, "failed to find the peer to stop rx aggregation\n");
-		spin_unlock_bh(&ab->data_lock);
+		spin_unlock_bh(&ab->base_lock);
 		return -ENOENT;
 	}
 
@@ -865,7 +865,7 @@ int ath11k_dp_rx_ampdu_stop(struct ath11
 
 	ath11k_peer_rx_tid_delete(ar, peer, params->tid);
 
-	spin_unlock_bh(&ab->data_lock);
+	spin_unlock_bh(&ab->base_lock);
 
 	if (!active)
 		return 0;
@@ -1123,11 +1123,11 @@ ath11k_update_per_peer_tx_stats(struct a
 	}
 
 	rcu_read_lock();
-	spin_lock_bh(&ab->data_lock);
+	spin_lock_bh(&ab->base_lock);
 	peer = ath11k_peer_find_by_id(ab, usr_stats->peer_id);
 
 	if (!peer || !peer->sta) {
-		spin_unlock_bh(&ab->data_lock);
+		spin_unlock_bh(&ab->base_lock);
 		rcu_read_unlock();
 		return;
 	}
@@ -1190,7 +1190,7 @@ ath11k_update_per_peer_tx_stats(struct a
 							    peer_stats, rate_idx);
 	}
 
-	spin_unlock_bh(&ab->data_lock);
+	spin_unlock_bh(&ab->base_lock);
 	rcu_read_unlock();
 }
 
@@ -1351,12 +1351,12 @@ static void ath11k_htt_backpressure_even
 		bp_stats = &ab->ring_stats.bp_stats.lmac_ring_bp_stats[ring_id][pdev_idx];
 	}
 
-	spin_lock_bh(&ab->data_lock);
+	spin_lock_bh(&ab->base_lock);
 	bp_stats->hp = hp;
 	bp_stats->tp = tp;
 	bp_stats->count++;
 	bp_stats->jiffies = jiffies;
-	spin_unlock_bh(&ab->data_lock);
+	spin_unlock_bh(&ab->base_lock);
 #endif
 }
 
@@ -2814,13 +2814,13 @@ int ath11k_dp_rx_process_mon_status(stru
 		}
 
 		rcu_read_lock();
-		spin_lock_bh(&ab->data_lock);
+		spin_lock_bh(&ab->base_lock);
 		peer = ath11k_peer_find_by_id(ab, ppdu_info.peer_id);
 
 		if (!peer || !peer->sta) {
 			ath11k_warn(ab, "failed to find the peer with peer_id %d\n",
 				    ppdu_info.peer_id);
-			spin_unlock_bh(&ab->data_lock);
+			spin_unlock_bh(&ab->base_lock);
 			rcu_read_unlock();
 			dev_kfree_skb_any(skb);
 			continue;
@@ -2834,7 +2834,7 @@ int ath11k_dp_rx_process_mon_status(stru
 			ath11k_rx_stats_buf_pktlog_process(ar, skb->data,
 							   DP_RX_BUFFER_SIZE);
 		}
-		spin_unlock_bh(&ab->data_lock);
+		spin_unlock_bh(&ab->base_lock);
 		rcu_read_unlock();
 
 		dev_kfree_skb_any(skb);
--- a/drivers/net/wireless/ath/ath11k/mac.c
+++ b/drivers/net/wireless/ath/ath11k/mac.c
@@ -682,13 +682,13 @@ void ath11k_mac_peer_cleanup_all(struct
 
 	lockdep_assert_held(&ar->conf_mutex);
 
-	spin_lock_bh(&ab->data_lock);
+	spin_lock_bh(&ab->base_lock);
 	list_for_each_entry_safe(peer, tmp, &ab->peers, list) {
 		ath11k_peer_rx_tid_cleanup(ar, peer);
 		list_del(&peer->list);
 		kfree(peer);
 	}
-	spin_unlock_bh(&ab->data_lock);
+	spin_unlock_bh(&ab->base_lock);
 
 	ar->num_peers = 0;
 	ar->num_stations = 0;
@@ -2227,11 +2227,11 @@ static int ath11k_clear_peer_keys(struct
 
 	lockdep_assert_held(&ar->conf_mutex);
 
-	spin_lock_bh(&ab->data_lock);
+	spin_lock_bh(&ab->base_lock);
 	peer = ath11k_peer_find(ab, arvif->vdev_id, addr);
 
 	if (!peer) {
-		spin_unlock_bh(&ab->data_lock);
+		spin_unlock_bh(&ab->base_lock);
 		return -ENOENT;
 	}
 
@@ -2240,7 +2240,7 @@ static int ath11k_clear_peer_keys(struct
 		peer->keys[i]= NULL;
 	}
 
-	spin_unlock_bh(&ab->data_lock);
+	spin_unlock_bh(&ab->base_lock);
 
 	for (i = 0; i < ARRAY_SIZE(keys); i++) {
 		if (!keys[i])
@@ -2300,9 +2300,9 @@ static int ath11k_mac_op_set_key(struct
 	/* the peer should not disappear in mid-way (unless FW goes awry) since
 	 * we already hold conf_mutex. we just make sure its there now.
 	 */
-	spin_lock_bh(&ab->data_lock);
+	spin_lock_bh(&ab->base_lock);
 	peer = ath11k_peer_find(ab, arvif->vdev_id, peer_addr);
-	spin_unlock_bh(&ab->data_lock);
+	spin_unlock_bh(&ab->base_lock);
 
 	if (!peer) {
 		if (cmd == SET_KEY) {
@@ -2329,7 +2329,7 @@ static int ath11k_mac_op_set_key(struct
 		goto exit;
 	}
 
-	spin_lock_bh(&ab->data_lock);
+	spin_lock_bh(&ab->base_lock);
 	peer = ath11k_peer_find(ab, arvif->vdev_id, peer_addr);
 	if (peer && cmd == SET_KEY)
 		peer->keys[key->keyidx] = key;
@@ -2338,7 +2338,7 @@ static int ath11k_mac_op_set_key(struct
 	else if (!peer)
 		/* impossible unless FW goes crazy */
 		ath11k_warn(ab, "peer %pM disappeared!\n", peer_addr);
-	spin_unlock_bh(&ab->data_lock);
+	spin_unlock_bh(&ab->base_lock);
 
 exit:
 	mutex_unlock(&ar->conf_mutex);
@@ -2911,17 +2911,17 @@ static void ath11k_mac_op_sta_rc_update(
 	struct ath11k_peer *peer;
 	u32 bw, smps;
 
-	spin_lock_bh(&ar->ab->data_lock);
+	spin_lock_bh(&ar->ab->base_lock);
 
 	peer = ath11k_peer_find(ar->ab, arvif->vdev_id, sta->addr);
 	if (!peer) {
-		spin_unlock_bh(&ar->ab->data_lock);
+		spin_unlock_bh(&ar->ab->base_lock);
 		ath11k_warn(ar->ab, "mac sta rc update failed to find peer %pM on vdev %i\n",
 			    sta->addr, arvif->vdev_id);
 		return;
 	}
 
-	spin_unlock_bh(&ar->ab->data_lock);
+	spin_unlock_bh(&ar->ab->base_lock);
 
 	ath11k_dbg(ar->ab, ATH11K_DBG_MAC,
 		   "mac sta rc update for %pM changed %08x bw %d nss %d smps %d\n",
@@ -4529,9 +4529,9 @@ ath11k_mac_vdev_start_restart(struct ath
 
 		arg.channel.passive = arg.channel.chan_radar;
 
-		spin_lock_bh(&ab->data_lock);
+		spin_lock_bh(&ab->base_lock);
 		arg.regdomain = ar->ab->dfs_region;
-		spin_unlock_bh(&ab->data_lock);
+		spin_unlock_bh(&ab->base_lock);
 
 		/* TODO: Notify if secondary 80Mhz also needs radar detection */
 		if (he_support) {
--- a/drivers/net/wireless/ath/ath11k/peer.c
+++ b/drivers/net/wireless/ath/ath11k/peer.c
@@ -12,7 +12,7 @@ struct ath11k_peer *ath11k_peer_find(str
 {
 	struct ath11k_peer *peer;
 
-	lockdep_assert_held(&ab->data_lock);
+	lockdep_assert_held(&ab->base_lock);
 
 	list_for_each_entry(peer, &ab->peers, list) {
 		if (peer->vdev_id != vdev_id)
@@ -31,7 +31,7 @@ struct ath11k_peer *ath11k_peer_find_by_
 {
 	struct ath11k_peer *peer;
 
-	lockdep_assert_held(&ab->data_lock);
+	lockdep_assert_held(&ab->base_lock);
 
 	list_for_each_entry(peer, &ab->peers, list) {
 		if (memcmp(peer->addr, addr, ETH_ALEN))
@@ -48,7 +48,7 @@ struct ath11k_peer *ath11k_peer_find_by_
 {
 	struct ath11k_peer *peer;
 
-	lockdep_assert_held(&ab->data_lock);
+	lockdep_assert_held(&ab->base_lock);
 
 	list_for_each_entry(peer, &ab->peers, list)
 		if (peer_id == peer->peer_id)
@@ -61,7 +61,7 @@ void ath11k_peer_unmap_event(struct ath1
 {
 	struct ath11k_peer *peer;
 
-	spin_lock_bh(&ab->data_lock);
+	spin_lock_bh(&ab->base_lock);
 
 	peer = ath11k_peer_find_by_id(ab, peer_id);
 	if (!peer) {
@@ -78,7 +78,7 @@ void ath11k_peer_unmap_event(struct ath1
 	wake_up(&ab->peer_mapping_wq);
 
 exit:
-	spin_unlock_bh(&ab->data_lock);
+	spin_unlock_bh(&ab->base_lock);
 }
 
 void ath11k_peer_map_event(struct ath11k_base *ab, u8 vdev_id, u16 peer_id,
@@ -86,7 +86,7 @@ void ath11k_peer_map_event(struct ath11k
 {
 	struct ath11k_peer *peer;
 
-	spin_lock_bh(&ab->data_lock);
+	spin_lock_bh(&ab->base_lock);
 	peer = ath11k_peer_find(ab, vdev_id, mac_addr);
 	if (!peer) {
 		peer = kzalloc(sizeof(*peer), GFP_ATOMIC);
@@ -105,7 +105,7 @@ void ath11k_peer_map_event(struct ath11k
 		   vdev_id, mac_addr, peer_id);
 
 exit:
-	spin_unlock_bh(&ab->data_lock);
+	spin_unlock_bh(&ab->base_lock);
 }
 
 static int ath11k_wait_for_peer_common(struct ath11k_base *ab, int vdev_id,
@@ -116,9 +116,9 @@ static int ath11k_wait_for_peer_common(s
 	ret = wait_event_timeout(ab->peer_mapping_wq, ({
 				bool mapped;
 
-				spin_lock_bh(&ab->data_lock);
+				spin_lock_bh(&ab->base_lock);
 				mapped = !!ath11k_peer_find(ab, vdev_id, addr);
-				spin_unlock_bh(&ab->data_lock);
+				spin_unlock_bh(&ab->base_lock);
 
 				(mapped == expect_mapped ||
 				 test_bit(ATH11K_FLAG_CRASH_FLUSH, &ab->dev_flags));
@@ -137,7 +137,7 @@ void ath11k_peer_cleanup(struct ath11k *
 
 	lockdep_assert_held(&ar->conf_mutex);
 
-	spin_lock_bh(&ab->data_lock);
+	spin_lock_bh(&ab->base_lock);
 	list_for_each_entry_safe(peer, tmp, &ab->peers, list) {
 		if (peer->vdev_id != vdev_id)
 			continue;
@@ -150,7 +150,7 @@ void ath11k_peer_cleanup(struct ath11k *
 		ar->num_peers--;
 	}
 
-	spin_unlock_bh(&ab->data_lock);
+	spin_unlock_bh(&ab->base_lock);
 }
 
 int ath11k_wait_for_peer_deleted(struct ath11k *ar, int vdev_id, const u8 *addr)
@@ -235,11 +235,11 @@ int ath11k_peer_create(struct ath11k *ar
 	if (ret)
 		return ret;
 
-	spin_lock_bh(&ar->ab->data_lock);
+	spin_lock_bh(&ar->ab->base_lock);
 
 	peer = ath11k_peer_find(ar->ab, param->vdev_id, param->peer_addr);
 	if (!peer) {
-		spin_unlock_bh(&ar->ab->data_lock);
+		spin_unlock_bh(&ar->ab->base_lock);
 		ath11k_warn(ar->ab, "failed to find peer %pM on vdev %i after creation\n",
 			    param->peer_addr, param->vdev_id);
 		ret = ath11k_wmi_send_peer_delete_cmd(ar, param->peer_addr,
@@ -262,7 +262,7 @@ int ath11k_peer_create(struct ath11k *ar
 
 	ar->num_peers++;
 
-	spin_unlock_bh(&ar->ab->data_lock);
+	spin_unlock_bh(&ar->ab->base_lock);
 
 	return 0;
 }
--- a/drivers/net/wireless/ath/ath11k/reg.c
+++ b/drivers/net/wireless/ath/ath11k/reg.c
@@ -202,7 +202,7 @@ int ath11k_regd_update(struct ath11k *ar
 	ab = ar->ab;
 	pdev_id = ar->pdev_idx;
 
-	spin_lock(&ab->data_lock);
+	spin_lock(&ab->base_lock);
 
 	if (init) {
 		/* Apply the regd received during init through
@@ -223,7 +223,7 @@ int ath11k_regd_update(struct ath11k *ar
 
 	if (!regd) {
 		ret = -EINVAL;
-		spin_unlock(&ab->data_lock);
+		spin_unlock(&ab->base_lock);
 		goto err;
 	}
 
@@ -234,7 +234,7 @@ int ath11k_regd_update(struct ath11k *ar
 	if (regd_copy)
 		ath11k_copy_regd(regd, regd_copy);
 
-	spin_unlock(&ab->data_lock);
+	spin_unlock(&ab->base_lock);
 
 	if (!regd_copy) {
 		ret = -ENOMEM;
--- a/drivers/net/wireless/ath/ath11k/wmi.c
+++ b/drivers/net/wireless/ath/ath11k/wmi.c
@@ -5322,7 +5322,7 @@ static int ath11k_reg_chan_list_event(st
 		goto fallback;
 	}
 
-	spin_lock(&ab->data_lock);
+	spin_lock(&ab->base_lock);
 	if (ab->mac_registered) {
 		/* Once mac is registered, ar is valid and all CC events from
 		 * fw is considered to be received due to user requests
@@ -5346,7 +5346,7 @@ static int ath11k_reg_chan_list_event(st
 		ab->default_regd[pdev_idx] = regd;
 	}
 	ab->dfs_region = reg_info->dfs_region;
-	spin_unlock(&ab->data_lock);
+	spin_unlock(&ab->base_lock);
 
 	goto mem_free;
 
@@ -5818,7 +5818,7 @@ static void ath11k_peer_sta_kickout_even
 
 	rcu_read_lock();
 
-	spin_lock_bh(&ab->data_lock);
+	spin_lock_bh(&ab->base_lock);
 
 	peer = ath11k_peer_find_by_addr(ab, arg.mac_addr);
 
@@ -5849,7 +5849,7 @@ static void ath11k_peer_sta_kickout_even
 	ieee80211_report_low_ack(sta, 10);
 
 exit:
-	spin_unlock_bh(&ab->data_lock);
+	spin_unlock_bh(&ab->base_lock);
 	rcu_read_unlock();
 }
 
--- a/drivers/net/wireless/ath/ath11k/dp_tx.c
+++ b/drivers/net/wireless/ath/ath11k/dp_tx.c
@@ -438,12 +438,12 @@ static void ath11k_dp_tx_complete_msdu(s
 	}
 
 
-	spin_lock_bh(&ab->data_lock);
+	spin_lock_bh(&ab->base_lock);
 	peer = ath11k_peer_find_by_id(ab, ts->peer_id);
 	if (!peer || !peer->sta) {
 		ath11k_warn(ab, "failed to find the peer with peer_id %d\n",
 			    ts->peer_id);
-		spin_unlock_bh(&ab->data_lock);
+		spin_unlock_bh(&ab->base_lock);
 		dev_kfree_skb_any(msdu);
 		goto exit;
 	}
@@ -454,7 +454,7 @@ static void ath11k_dp_tx_complete_msdu(s
 	status.rate = &arsta->last_txrate;
 	rcu_read_unlock();
 	ieee80211_tx_status_ext(ar->hw, &status);
-	spin_unlock_bh(&ab->data_lock);
+	spin_unlock_bh(&ab->base_lock);
 	return;
 
 exit:
--- a/drivers/net/wireless/ath/ath11k/debug.c
+++ b/drivers/net/wireless/ath/ath11k/debug.c
@@ -840,7 +840,7 @@ static int ath11k_fill_bp_stats(struct a
 				struct ath11k_bp_stats *bp_stats,
 				char *buf, int len, int size)
 {
-	lockdep_assert_held(&ab->data_lock);
+	lockdep_assert_held(&ab->base_lock);
 
 	len += scnprintf(buf + len, size - len, "count: %u\n",
 			 bp_stats->count);
@@ -864,7 +864,7 @@ int ath11k_fill_bp_stats_buf(struct ath1
 	len += scnprintf(buf + len, size - len, "Backpressure Stats\n");
 	len += scnprintf(buf + len, size - len, "==================\n");
 
-	spin_lock_bh(&ab->data_lock);
+	spin_lock_bh(&ab->base_lock);
 	for (i = 0; i < HTT_SW_UMAC_RING_IDX_MAX; i++) {
 		bp_stats = &ab->ring_stats.bp_stats.umac_ring_bp_stats[i];
 
@@ -892,7 +892,7 @@ int ath11k_fill_bp_stats_buf(struct ath1
 			stats_rxd = true;
 		}
 	}
-	spin_unlock_bh(&ab->data_lock);
+	spin_unlock_bh(&ab->base_lock);
 
 	if (!stats_rxd)
 		len += scnprintf(buf + len, size - len, "No Ring Backpressure stats received\n\n");
